File: fsharp/raindrops/Raindrops.fs
module Raindrops

let convert (number: int): string = 
    let result = ""
    let pling = 
        if number % 3 = 0 then "Pling" else ""
    let plang = 
        if number % 5 = 0 then "Plang" else ""
    let plong = 
        if number % 7 = 0 then "Plong" else ""

    let drops = pling + plang + plong
    if drops = "" 
        then string number
        else drops

File: c/raindrops/raindrops.c
#include "raindrops.h"

void convert(char result[], int drops) {
    char *pling = drops % 3 == 0 ? "Pling" : "";
    char *plang = drops % 5 == 0 ? "Plang" : "";
    char *plong = drops % 7 == 0 ? "Plong" : "";
    sprintf(result, "%s%s%s", pling, plang, plong);
    if (strlen(result) == 0) sprintf(result, "%d", drops);
}
